File: buildinfo/buildinfo.go
// Package buildinfo exposes build metadata injected through linker flags.
package buildinfo

import "fmt"

var (
	// Version is the release version.
	Version = "development"
	// Commit is the source revision.
	Commit = "unknown"
	// BuildTime is the RFC 3339 build timestamp.
	BuildTime = "unknown"
)

// String returns stable, human-readable build metadata.
func String() string {
	return fmt.Sprintf("tidekeepers-api version=%s commit=%s built=%s", Version, Commit, BuildTime)
}

File: src/buildinfo.ts
/** Build metadata equivalent to the Go buildinfo package. */

export interface BuildInfo {
  version: string;
  commit: string;
  buildTime: string;
}

export function buildInfo(
  lookup: (name: string) => string | undefined = (name) => Deno.env.get(name),
): BuildInfo {
  return {
    version: lookup("SERVICE_VERSION")?.trim() || "development",
    commit: lookup("GIT_COMMIT")?.trim() || "unknown",
    buildTime: lookup("BUILD_TIME")?.trim() || "unknown",
  };
}

export function buildInfoString(info = buildInfo()): string {
  return `tidekeepers-api version=${info.version} commit=${info.commit} built=${info.buildTime}`;
}
